grails-app/controllers/org/bbop/apollo/OrganismController.groovy

```diff
@@ -29,32 +33,66 @@ class OrganismController {
     def preferenceService
     def organismService
     def reportService
+    def configWrapperService
+    def trackService
 
 
     @RestApiMethod(description = "Remove an organism", path = "/organism/deleteOrganism", verb = RestApiVerb.POST)
     @RestApiParams(params = [
             @RestApiParam(name = "username", type = "email", paramType = RestApiParamType.QUERY)
             , @RestApiParam(name = "password", type = "password", paramType = RestApiParamType.QUERY)
-            , @RestApiParam(name = "organism", type = "json", paramType = RestApiParamType.QUERY, description = "Pass an Organism JSON object with an 'id' that corresponds to the id to delete")
+            , @RestApiParam(name = "organism", type = "string", paramType = RestApiParamType.QUERY, description = "ID or commonName that can be used to uniquely identify an organism")
+            , @RestApiParam(name = "deleteData", type = "boolean", paramType = RestApiParamType.QUERY, description = "Delete organism's data directory (applicable only to those organisms with its data added via web services)")
     ])
     @Transactional
     def deleteOrganism() {
+        /*
+            Representative CURL query:
+
+            curl http://localhost:8080/apollo/organism/deleteOrganism  \
+            -F "organism=Amel" \
+            -F "username=admin" \
+            -F "password=admin" \
+            -F "deleteData=true" \
+            -X POST
+         */
         try {
-            JSONObject organismJson = permissionService.handleInput(request, params)
-            log.debug "deleteOrganism ${organismJson}"
-            if (permissionService.isUserAdmin(permissionService.getCurrentUser(organismJson))) {
+            JSONObject requestObject = permissionService.handleInput(request, params)
+            log.debug "deleteOrganism ${requestObject}"
+            if (permissionService.isUserAdmin(permissionService.getCurrentUser(requestObject))) {
+
+                log.debug "organism ID: ${requestObject.organism}"
+                Organism organism = preferenceService.getOrganismForTokenInDB(requestObject.organism)
 
-                log.debug "organism ID: ${organismJson.id}"
-                Organism organism = Organism.findById(organismJson.id as Long)
-                if (!organism) {
-                    organism = Organism.findByCommonName(organismJson.organism)
-                }
                 if (organism) {
+                    boolean addedViaWebServices = organism.addedViaWebServices
```

Comment:
I think should be:

     organism.addedViaWebServices==null ? false : organism.addedViaWebServices

but maybe this works fine.   Would just double-check if you add organisms the regular way that it works.